### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by joel-kalema**
- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open rails console for your project.
- Create at least one user
- Create at least 4 posts written by one of the users you created
- Create at least 6 posts comments for one of the posts you created
- 
 ## Your models should include some additional methods.
- [ ] Users
- A method that returns the 3 most recent posts for a given user.
- [ ] Posts
 - A method that updates the posts counter for a user.
  A method which returns the 5 most recent comments for a given post.
- [ ] Comments
 - A method that updates the comments counter for a post.
- [ ] Likes
- A method that updates the likes counter for a post.
- Go to rails c and check if your methods are working.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class User < ApplicationRecord
+  has_many :posts, class_name: 'post', foreign_key: 'author_id'
+  has_many :comments, class_name: 'comment', foreign_key: 'author_id'
+  has_many :likes, class_name: 'like', foreign_key: 'author_id'
```
Comment (gemmen29) [inline app/models/user.rb:4]: - [ ] Kindly identify the method that returns the 3 most recent post for a given user as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#:~:text=A%20method%20that%20returns%20the%203%20most%20recent%20posts%20for%20a%20given%20user.).

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,5 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, class_name: 'Comment', foreign_key: 'post_id'
+  has_many :likes, class_name: 'Like', foreign_key: 'post_id'
```
Comment (gemmen29) [inline app/models/post.rb:4]: - [ ] Kindly identify a method that updates the posts counter for a user as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#:~:text=A%20method%20that%20updates%20the%20posts%20counter%20for%20a%20user.).
- [ ] Kindly identify A method that returns the 5 most recent comments for a given post as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#:~:text=A%20method%20which%20returns%20the%205%20most%20recent%20comments%20for%20a%20given%20post.).

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,2 +1,9 @@
 class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+  after_save :update_comments_counter
+
+  # def update_comments_counter
+  #   post.increment!(:comments_counter)
+  # end
```
Comment (gemmen29) [inline app/models/comment.rb]: - [ ] Kindly uncomment this code as this is correct to implement this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#:~:text=A%20method%20that%20updates%20the%20comments%20counter%20for%20a%20post.).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @joel-kalema, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @joel-kalema 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Great work addressing the requested changes from the previous reviewer correctly 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**